src/rascal/structure_managers/structure_manager.hh

```diff
@@ -320,72 +325,114 @@ namespace rascal {
     }
 
     /**
-     * Attach a property to a StructureManager. A given calculated property is
-     * only reasonable if connected with a structure. It is also connected with
-     * a sanity check so that the naming of attached properties is unique. If a
-     * property with the desired `name` already exists, a runtime error is
-     * thrown.
+     * Helper function to check if a property with the specifier `name` has
+     * already been attached in the manager layer which invoked this function.
      */
-    void attach_property(const std::string & name,
-                         std::shared_ptr<PropertyBase> property) {
-      if (this->has_property(name)) {
-        std::stringstream error{};
-        error << "A property of name '" << name
-              << "' has already been registered";
-        throw std::runtime_error(error.str());
-      }
-      this->properties[name] = property;
+    inline bool has_layer_property(const std::string & name) const {
+      return not(this->properties.find(name) == this->properties.end());
     }
 
     /**
      * Helper function to check if a property with the specifier `name` has
-     * already been attached.
+     * already been attached somewhere in the manager stack. Here an example how
+     * it works
+     *
+     * Property request forwarding in the case property exists
+     * AdaptorImpl2 -> has not prop1
+     *       | forwards request to lower stack
+     *       v
+     * AdaptorImpl1 has prop1 -> return true
+     * RootImpl
+     *
+     *
+     * Property request forwarding in the case property does not exist
+     * AdaptorImpl2
+     *       | forwards request to lower stack
+     *       v
+     * AdaptorImpl1
+     *       | forwards request to lower stack
+     *       v
+     * RootImpl -> return false
      */
-    bool has_property(const std::string & name) const {
-      return not(this->properties.find(name) == this->properties.end());
-    }
-
-    template <typename Property_t>
-    void create_property(const std::string & name) {
-      auto property{std::make_shared<Property_t>(this->implementation())};
-      this->attach_property(name, property);
+    template <bool IsRoot = IsRootImplementation,
+              std::enable_if_t<IsRoot, int> = 0>
+    inline bool has_stack_property(const std::string & name) {
```

Comment:
Maybe `has_property_in_current_stack` ?

Reply:
To be clear
AdatporNL < StructureManagerCenter > is the whole stack
AdatporNL is a layer and StructureManagerCenter is a layer.

So `has_stack_property` looks in the whole stack not just the top layer. I could rename it to `has_whole_stack_property`

Reply:
`is_property_in_stack`